app/frontend/state.ts

```diff
@@ -80,11 +84,24 @@ export interface State {
 
   // transaction
   sendTransactionSummary: TransactionSummary
+
+  // Note: This allows for multiple transaction summaries by TxType, to avoid race-conditions
+  // when calculating transaction summaries for more TxTypes at the same time.
+  // Note that this still does not allow for multiple cached transaction summaries from
+  // the same TxType, however, this should not be use-case for us in the near future.
+  cachedTransactionSummaries: {
+    [TxType.CONVERT_LEGACY]?: TransactionSummary & SendTransactionSummary
```

Comment:
Any ideas on how to make this maybe "better" but not overcomplicate it? Or do you consider this fine?

Reply:
I would prefer another PR, to close this quicker. I will create an issue for that.